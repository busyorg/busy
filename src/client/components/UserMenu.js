import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Scrollbars } from 'react-custom-scrollbars';
import { FormattedMessage, FormattedNumber } from 'react-intl';
import './UserMenu.less';

class UserMenu extends React.Component {
  static propTypes = {
    onChange: PropTypes.func,
    defaultKey: PropTypes.string,
    followers: PropTypes.number,
    following: PropTypes.number,
  };

  static defaultProps = {
    onChange: () => {},
    defaultKey: 'discussions',
    followers: 0,
    following: 0,
  };

  constructor(props) {
    super(props);
    this.state = {
      current: props.defaultKey ? props.defaultKey : 'discussions',
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      current: nextProps.defaultKey ? nextProps.defaultKey : 'discussions',
    });
  }

  getItemClasses = key =>
    classNames('UserMenu__item', { 'UserMenu__item--active': this.state.current === key });

  handleClick = e => {
    const key = e.currentTarget.dataset.key;
    this.setState({ current: key }, () => this.props.onChange(key));
  };

  render() {
    return (
      <div className="UserMenu">
        <div className="container menu-layout">
          <div className="left" />
          <Scrollbars universal autoHide style={{ width: '100%', height: 46 }}>
            <ul className="UserMenu__menu center">
              <li
                className={this.getItemClasses('discussions')}
                onClick={this.handleClick}
                role="presentation"
                data-key="discussions"
              >
                <FormattedMessage id="discussions" defaultMessage="Discussions" />
              </li>
              <li
                className={this.getItemClasses('comments')}
                onClick={this.handleClick}
                role="presentation"
                data-key="comments"
              >
                <FormattedMessage id="comments" defaultMessage="Comments" />
              </li>
              <li
                className={this.getItemClasses('followers')}
                onClick={this.handleClick}
                role="presentation"
                data-key="followers"
              >
                <FormattedMessage id="followers" defaultMessage="Followers" />
                <span className="UserMenu__badge">
                  <FormattedNumber value={this.props.followers} />
                </span>
              </li>
              <li
                className={this.getItemClasses('followed')}
                onClick={this.handleClick}
                role="presentation"
                data-key="followed"
              >
                <FormattedMessage id="following" defaultMessage="Following" />
                <span className="UserMenu__badge">
                  <FormattedNumber value={this.props.following} />
                </span>
              </li>
              <li
                className={this.getItemClasses('transfers')}
                onClick={this.handleClick}
                role="presentation"
                data-key="transfers"
              >
                <FormattedMessage id="wallet" defaultMessage="Wallet" />
              </li>
              <li
                className={this.getItemClasses('activity')}
                onClick={this.handleClick}
                role="presentation"
                data-key="activity"
              >
                <FormattedMessage id="activity" defaultMessage="Activity" />
              </li>
            </ul>
          </Scrollbars>
        </div>
      </div>
    );
  }
}

export default UserMenu;
