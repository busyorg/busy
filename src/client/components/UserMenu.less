@import (reference) '../styles/custom.less';

.UserMenu {
  height: 46px;
  background-color: white;
  border-top: solid 1px #e9e7e7;
  border-bottom: solid 1px #e9e7e7;

  &__menu {
    height: 46px;
    line-height: 46px;
    white-space: nowrap;
  }

  &__item {
    display: inline-block;
    height: 45px;
    margin-right: 16px;
    cursor: pointer;
    color: #99aab5;
    font-weight: bold;

    &--active,
    &:hover {
      color: @purple;

      border-bottom: 2px solid @purple;
    }
  }

  &__badge {
    color: #c2ccd3;
    margin-left: 6px;
  }
}

div[data-dir='rtl'] .UserMenu {
  &__badge {
    margin-left: 0px;
    margin-right: 6px;
  }

  &__item {
    margin-right: 0;
    margin-left: 16px;
  }
}
