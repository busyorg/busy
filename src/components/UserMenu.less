@import "../styles/custom.less";

.UserMenu {
  background: @white;
  border-top: solid 1px #e9e7e7;
  border-bottom: solid 1px #e9e7e7;

  &__menu {
    margin: 0 auto;
    max-width: 990px;
    border: none !important;
  }

  &__item {
    color: #99aab5;
    font-weight: bold;
  }

  &__badge {
    color: #c2ccd3;
    margin-left: 8px;
  }
}
