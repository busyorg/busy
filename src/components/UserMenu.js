import React from 'react';
import PropTypes from 'prop-types';
import numeral from 'numeral';
import { Menu } from 'antd';
import './UserMenu.less';

class UserMenu extends React.Component {
  static propTypes = {
    onChange: PropTypes.func,
    defaultKey: PropTypes.string,
    discussions: PropTypes.number,
    comments: PropTypes.number,
    followers: PropTypes.number,
    following: PropTypes.number,
  };

  static defaultProps = {
    onChange: () => {},
    defaultKey: 'discussions',
    discussions: 0,
    comments: 0,
    followers: 0,
    following: 0,
  };

  constructor(props) {
    super(props);
    this.state = {
      current: props.defaultKey ? props.defaultKey : 'discussions',
    };
  }

  handleClick = (e) => {
    this.setState(
      {
        current: e.key,
      },
      () => {
        if (this.props.onChange) {
          this.props.onChange(e.key);
        }
      },
    );
  };

  render() {
    const discussionsValue = numeral(this.props.discussions).format('0,0');
    const commentsValue = numeral(this.props.comments).format('0,0');
    const followersValue = numeral(this.props.followers).format('0,0');
    const followingValue = numeral(this.props.following).format('0,0');

    return (
      <div className="UserMenu">
        <div className="container topnav-layout">
          <div className="left" />
          <Menu
            className="UserMenu__menu center"
            onClick={this.handleClick}
            selectedKeys={[this.state.current]}
            mode="horizontal"
          >
            <Menu.Item key="discussions" className="UserMenu__item">
              Discussions
              <span className="UserMenu__badge">{discussionsValue}</span>
            </Menu.Item>
            <Menu.Item key="comments" className="UserMenu__item">
              <strong>
                Comments
                <span className="UserMenu__badge">{commentsValue}</span>
              </strong>
            </Menu.Item>
            <Menu.Item key="followers" className="UserMenu__item">
              <strong>
                Followers
                <span className="UserMenu__badge">{followersValue}</span>
              </strong>
            </Menu.Item>
            <Menu.Item key="followed" className="UserMenu__item">
              <strong>
                Following
                <span className="UserMenu__badge">{followingValue}</span>
              </strong>
            </Menu.Item>
          </Menu>
        </div>
      </div>
    );
  }
}

export default UserMenu;
