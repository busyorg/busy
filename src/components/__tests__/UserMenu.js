import React from 'react';
import { shallow } from 'enzyme';
import UserMenu from '../UserMenu';

describe('<UserMenu />', () => {
  it('renders without exploding', () => {
    const props = {
      defaultKey: 'comments',
      followers: 42,
      following: 21,
    };
    const wrapper = shallow(<UserMenu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
