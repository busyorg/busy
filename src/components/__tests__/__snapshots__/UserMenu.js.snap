// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenu /> renders without exploding 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="UserMenu"
  >
    <div
      className="container menu-layout"
    >
      <div
        className="left"
      />
      <Scrollbars
        autoHeight={false}
        autoHeightMax={200}
        autoHeightMin={0}
        autoHide={true}
        autoHideDuration={200}
        autoHideTimeout={1000}
        hideTracksWhenNotNeeded={false}
        renderThumbHorizontal={[Function]}
        renderThumbVertical={[Function]}
        renderTrackHorizontal={[Function]}
        renderTrackVertical={[Function]}
        renderView={[Function]}
        style={
          Object {
            "height": 46,
            "width": "100%",
          }
        }
        tagName="div"
        thumbMinSize={30}
        universal={false}
      >
        <ul
          className="UserMenu__menu center"
        >
          <li
            className="UserMenu__item"
            data-key="discussions"
            onClick={[Function]}
            role="presentation"
          >
            <FormattedMessage
              defaultMessage="Discussions"
              id="discussions"
              values={Object {}}
            />
          </li>
          <li
            className="UserMenu__item UserMenu__item--active"
            data-key="comments"
            onClick={[Function]}
            role="presentation"
          >
            <FormattedMessage
              defaultMessage="Comments"
              id="comments"
              values={Object {}}
            />
          </li>
          <li
            className="UserMenu__item"
            data-key="followers"
            onClick={[Function]}
            role="presentation"
          >
            <FormattedMessage
              defaultMessage="Followers"
              id="followers"
              values={Object {}}
            />
            <span
              className="UserMenu__badge"
            >
              <FormattedNumber
                value={42}
              />
            </span>
          </li>
          <li
            className="UserMenu__item"
            data-key="followed"
            onClick={[Function]}
            role="presentation"
          >
            <FormattedMessage
              defaultMessage="Following"
              id="following"
              values={Object {}}
            />
            <span
              className="UserMenu__badge"
            >
              <FormattedNumber
                value={21}
              />
            </span>
          </li>
          <li
            className="UserMenu__item"
            data-key="transfers"
            onClick={[Function]}
            role="presentation"
          >
            <FormattedMessage
              defaultMessage="Wallet"
              id="wallet"
              values={Object {}}
            />
          </li>
        </ul>
      </Scrollbars>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="UserMenu"
    >
      <div
        className="container menu-layout"
      >
        <div
          className="left"
        />
        <Scrollbars
          autoHeight={false}
          autoHeightMax={200}
          autoHeightMin={0}
          autoHide={true}
          autoHideDuration={200}
          autoHideTimeout={1000}
          hideTracksWhenNotNeeded={false}
          renderThumbHorizontal={[Function]}
          renderThumbVertical={[Function]}
          renderTrackHorizontal={[Function]}
          renderTrackVertical={[Function]}
          renderView={[Function]}
          style={
            Object {
              "height": 46,
              "width": "100%",
            }
          }
          tagName="div"
          thumbMinSize={30}
          universal={false}
        >
          <ul
            className="UserMenu__menu center"
          >
            <li
              className="UserMenu__item"
              data-key="discussions"
              onClick={[Function]}
              role="presentation"
            >
              <FormattedMessage
                defaultMessage="Discussions"
                id="discussions"
                values={Object {}}
              />
            </li>
            <li
              className="UserMenu__item UserMenu__item--active"
              data-key="comments"
              onClick={[Function]}
              role="presentation"
            >
              <FormattedMessage
                defaultMessage="Comments"
                id="comments"
                values={Object {}}
              />
            </li>
            <li
              className="UserMenu__item"
              data-key="followers"
              onClick={[Function]}
              role="presentation"
            >
              <FormattedMessage
                defaultMessage="Followers"
                id="followers"
                values={Object {}}
              />
              <span
                className="UserMenu__badge"
              >
                <FormattedNumber
                  value={42}
                />
              </span>
            </li>
            <li
              className="UserMenu__item"
              data-key="followed"
              onClick={[Function]}
              role="presentation"
            >
              <FormattedMessage
                defaultMessage="Following"
                id="following"
                values={Object {}}
              />
              <span
                className="UserMenu__badge"
              >
                <FormattedNumber
                  value={21}
                />
              </span>
            </li>
            <li
              className="UserMenu__item"
              data-key="transfers"
              onClick={[Function]}
              role="presentation"
            >
              <FormattedMessage
                defaultMessage="Wallet"
                id="wallet"
                values={Object {}}
              />
            </li>
          </ul>
        </Scrollbars>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserMenu
        defaultKey="comments"
        followers={42}
        following={21}
        onChange={[Function]}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "getItemClasses": [Function],
        "handleClick": [Function],
        "props": Object {
          "defaultKey": "comments",
          "followers": 42,
          "following": 21,
          "onChange": [Function],
        },
        "refs": Object {},
        "state": Object {
          "current": "comments",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="UserMenu"
        >
          <div
            className="container menu-layout"
          >
            <div
              className="left"
            />
            <Scrollbars
              autoHeight={false}
              autoHeightMax={200}
              autoHeightMin={0}
              autoHide={true}
              autoHideDuration={200}
              autoHideTimeout={1000}
              hideTracksWhenNotNeeded={false}
              renderThumbHorizontal={[Function]}
              renderThumbVertical={[Function]}
              renderTrackHorizontal={[Function]}
              renderTrackVertical={[Function]}
              renderView={[Function]}
              style={
                Object {
                  "height": 46,
                  "width": "100%",
                }
              }
              tagName="div"
              thumbMinSize={30}
              universal={false}
            >
              <ul
                className="UserMenu__menu center"
              >
                <li
                  className="UserMenu__item"
                  data-key="discussions"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Discussions"
                    id="discussions"
                    values={Object {}}
                  />
                </li>
                <li
                  className="UserMenu__item UserMenu__item--active"
                  data-key="comments"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Comments"
                    id="comments"
                    values={Object {}}
                  />
                </li>
                <li
                  className="UserMenu__item"
                  data-key="followers"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Followers"
                    id="followers"
                    values={Object {}}
                  />
                  <span
                    className="UserMenu__badge"
                  >
                    <FormattedNumber
                      value={42}
                    />
                  </span>
                </li>
                <li
                  className="UserMenu__item"
                  data-key="followed"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Following"
                    id="following"
                    values={Object {}}
                  />
                  <span
                    className="UserMenu__badge"
                  >
                    <FormattedNumber
                      value={21}
                    />
                  </span>
                </li>
                <li
                  className="UserMenu__item"
                  data-key="transfers"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Wallet"
                    id="wallet"
                    values={Object {}}
                  />
                </li>
              </ul>
            </Scrollbars>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="UserMenu"
        >
          <div
            className="container menu-layout"
          >
            <div
              className="left"
            />
            <Scrollbars
              autoHeight={false}
              autoHeightMax={200}
              autoHeightMin={0}
              autoHide={true}
              autoHideDuration={200}
              autoHideTimeout={1000}
              hideTracksWhenNotNeeded={false}
              renderThumbHorizontal={[Function]}
              renderThumbVertical={[Function]}
              renderTrackHorizontal={[Function]}
              renderTrackVertical={[Function]}
              renderView={[Function]}
              style={
                Object {
                  "height": 46,
                  "width": "100%",
                }
              }
              tagName="div"
              thumbMinSize={30}
              universal={false}
            >
              <ul
                className="UserMenu__menu center"
              >
                <li
                  className="UserMenu__item"
                  data-key="discussions"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Discussions"
                    id="discussions"
                    values={Object {}}
                  />
                </li>
                <li
                  className="UserMenu__item UserMenu__item--active"
                  data-key="comments"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Comments"
                    id="comments"
                    values={Object {}}
                  />
                </li>
                <li
                  className="UserMenu__item"
                  data-key="followers"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Followers"
                    id="followers"
                    values={Object {}}
                  />
                  <span
                    className="UserMenu__badge"
                  >
                    <FormattedNumber
                      value={42}
                    />
                  </span>
                </li>
                <li
                  className="UserMenu__item"
                  data-key="followed"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Following"
                    id="following"
                    values={Object {}}
                  />
                  <span
                    className="UserMenu__badge"
                  >
                    <FormattedNumber
                      value={21}
                    />
                  </span>
                </li>
                <li
                  className="UserMenu__item"
                  data-key="transfers"
                  onClick={[Function]}
                  role="presentation"
                >
                  <FormattedMessage
                    defaultMessage="Wallet"
                    id="wallet"
                    values={Object {}}
                  />
                </li>
              </ul>
            </Scrollbars>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserMenu
    defaultKey="comments"
    followers={42}
    following={21}
    onChange={[Function]}
  />,
}
`;
